server.js

```diff
@@ -0,0 +1,104 @@
+if (process.env.NODE_ENV !== 'production') {
+  require('dotenv').config()
+}
+
+const express = require('express')
+const app = express()
+const bcrypt = require('bcrypt')
+const passport = require('passport')
+const flash = require('express-flash')
+const session = require('express-session')
+const methodOverride = require('method-override')
+const server = require('http').createServer(app);
+const WebSocket = require('ws');
+const initializePassport = require('./passport-config')
+initializePassport(
+  passport,
+  email => users.find(user => user.email === email),
+  id => users.find(user => user.id === id)
+)
+
+const users = []
+
+const wss = new WebSocket.Server({server});
+
+app.set('view-engine', 'ejs')
+app.use(express.urlencoded({ extended: false }))
+app.use(flash())
+app.use(session({
+  secret: process.env.SESSION_SECRET,
+  resave: true,
+  saveUninitialized: true
+}))
+app.use(passport.initialize())
+app.use(passport.session())
+app.use(methodOverride('_method'))
+
+wss.on('connection', function connection(ws){
+  console.log('A new client Connection!')
+  ws.send('Welcome new Client');
+
+  ws.on('message', function incoming(message){
+    console.log('received: %s', message);
+    ws.send('Got your message its :' + message)
+  });
+});
+
+app.get('/', (req, res) => res.send('Hello World!'))
+
+server.listen(3000, () => console.log('Listenining on port : 3000'))
+
+// app.get('/', checkAuthenticated, (req, res) => {
+//   res.render('index.ejs', { name: req.user.name })
+// })
+
+app.get('/login', checkNotAuthenticated, (req, res) => {
+  res.render('login.ejs')
+})
+
+app.post('/login', checkNotAuthenticated, passport.authenticate('local', {
+  successRedirect: '/',
+  failureRedirect: '/login',
+  failureFlash: true
+}))
+
+app.get('/register', checkNotAuthenticated, (req, res) => {
+  res.render('register.ejs')
+})
+
+app.post('/register', checkNotAuthenticated, async (req, res) => {
+  try {
+    const hashedPassword = await bcrypt.hash(req.body.password, 10)
+    users.push({
+      id: Date.now().toString(),
+      name: req.body.name,
+      email: req.body.email,
+      password: hashedPassword
+    })
+    res.redirect('/login')
```

Comment:
Users should be added to a relational database (at least SQLite).

Reply:
Added mongodb to store users, used docker to run mongodb server.

Use below command - docker run --name mongodb -d -p 27017:27017 mongo

Reply:
Code reference - https://javascript.plainenglish.io/jwt-auth-with-node-and-passport-js-c41a91d333e0